infra/modules/database/role-manager.tf

```diff
@@ -94,26 +104,43 @@ resource "aws_iam_role" "role_manager" {
   ]
 }
 
-data "aws_secretsmanager_secret" "db_pass" {
-  arn = aws_rds_cluster.db.master_user_secret[0].secret_arn
-}
+
 
 resource "aws_iam_role_policy" "ssm_access" {
   name = "${var.name}-role-manager-ssm-access"
   role = aws_iam_role.role_manager.id
 
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = ["kms:Decrypt"]
+        Resource = [data.aws_kms_key.default_ssm_key.arn]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "database_credential_tool" {
+  count = length(aws_rds_cluster.db.master_user_secret)
```

Comment:
@charles-nava @shawnvanderjagt I don't think we want the count here. See https://github.com/navapbc/template-infra/issues/475 as discovered by @rocketnova